### DISCUSSION TITLE
ARM ipxe to load ARM WinPE

### DISCUSSION THREAD

**Discussion by amdhugo** [ipxe/ipxe#1364] [General]
I compile a iPXE for ARM via this command from latest ipxe repo,
**_make CROSS_COMPILE=aarch64-linux-gnu- bin-arm64-efi/ipxe.efi EMBED=fix.ipxe -j 16_**

I can boot into it. I am trying to load a ARM winpe which I create it by this command in Windows Kits(10.1.26100.2454) environment,
**_copype arm64 c:\winpe_arm_**
I get the **_BCD and boot.sdi_** files from c:\winpe_arm\media\EFI\Microsoft\Boot and c:\winpe_arm\media\boot.sdi
However, I got error to load it in arm ipxe environment by this way,
initrd ${base-url}/BCD  BCD
initrd ${base-url}/boot.sdi     boot.sdi
initrd ${base-url}/winpe-sample/boot.wim     boot.wim
boot || goto failed

Refer to error screen capture. Is that any suggestion on how to resolve it?
![image](https://github.com/user-attachments/assets/4f60a344-314f-4b0d-8664-d978b617d8f1)

**Comment by NiKiZe** [comment]
WAIK removes files from boot.wim that you will have to provide externally, so grab official iso and try boot.wim from there first.

But is the wimboot binary the arm64 version?

The error message shows that iPXE could not boot https://ipxe.org/7f048183, which is the ego firmware couldn't boot the binary.
Run `imgstat` before `boot` and post output. If it shows that wimboot is x86 efi, then you got the wrong file.

**Reply by amdhugo** [reply]
I might make sense. I was not using the right wimboot. After I use a correct one, then everything goes well.
